### PR TITLE
Add code coverage exclusions on false positives

### PR DESCRIPTION
**Pull request by peternied**
### Description
Adding code coverage exclusions on files that are creating code coverage measurement issues.

### Issues Resolved
- Related https://github.com/opensearch-project/security/issues/3137

### Testing
Will be checking the codecov output on this PR and expect it to come back clean.  Might need to iterate a couple times to make sure the max threshold % amount is correct.

### Check List
- [ ] ~New functionality includes testing~
- [ ] ~New functionality has been documented~
- [X] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: codecov.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+coverage:
+  status:
+    project:
+      default:
```
Comment (peternied) [inline codecov.yml:4]: Hmm, isn't clear that its putting default into one bucket and these files into another, need to look into this...

File: codecov.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        target: auto
+        threshold: 0%
+      # Exclusions to prevent false positives triggered
+      # Tracking with https://github.com/opensearch-project/security/issues/3137
+      inconsistant-coverage-files:
```
Comment (cwperks) [inline codecov.yml:9]: nit: inconsistent

File: codecov.yml (hunk 1 above)
Comment (peternied) [reply codecov.yml]: Looks like the exclusions must be manually declared, was able to see this by looking at the coverage configuration at the commit level - https://app.codecov.io/gh/opensearch-project/security/pull/3196/commits

Comment (peternied) [reply codecov.yml]: I've resolved this by seperating the plugin out with its own exclusions which is working as intented

image

**Review by DarshitChanpura — APPROVED** [review]

**Review by cwperks — APPROVED** [review]
@peternied I'm approving this PR, but also had a question. Could this be achieved using java annotations?

**Comment by peternied** [comment]
> Could this be achieved using java annotations?

Codecov doesn't support any java annotations that I'm aware of.  The only really annotations that I've seen used in the code coverage space are about automatically excluding code that was generated, such as by adding your own @.generated annotation [example](https://www.baeldung.com/jacoco-report-exclude#excluding-with-custom-annotation).

**Pull request merged**